src/react-imgix.jsx

```diff
@@ -124,25 +124,28 @@ function buildSrc({
       srcSet = `${dpr1} 1x, ${dpr2} 2x, ${dpr3} 3x, ${dpr4} 4x, ${dpr5} 5x`;
     } else {
       let showARWarning = false;
+      const { width, height, ...urlParams } = srcOptions;
+      const constructedUrl = constructUrl(rawSrc, urlParams);
+
+      const aspectRatioDecimal = parseAspectRatio(aspectRatio);
+      if (aspectRatio != null && aspectRatioDecimal === false) {
+        // false indicates invalid
+        showARWarning = true;
+      }
+
       const buildSrcSetPair = targetWidth => {
```

Comment:
Optional: this function could be moved out into the module scope, and the variables used passed in as parameters.

Reply:
If this is done, I'm also curious as to what effect memoization could have on the performance of this function. My guess is that it wouldn't improve performance... but miracles can happen.